org.adempiere.base/src/org/compiere/model/MCost.java

```diff
@@ -1428,6 +1498,40 @@ public static MCost get (Properties ctx, int AD_Client_ID, int AD_Org_ID, int M_
 		int M_AttributeSetInstance_ID,
 		String trxName)
 	{
+		return get (ctx, AD_Client_ID, AD_Org_ID, M_Product_ID, M_CostType_ID, C_AcctSchema_ID, M_CostElement_ID, M_AttributeSetInstance_ID, null, null, trxName);
+	}
+	
+	/**
+	 * Get Cost Record
+	 * @param ctx context
+	 * @param AD_Client_ID client
+	 * @param AD_Org_ID org
+	 * @param M_Product_ID product
+	 * @param M_CostType_ID cost type
+	 * @param C_AcctSchema_ID as
+	 * @param M_CostElement_ID cost element
+	 * @param M_AttributeSetInstance_ID asi
+	 * @param dateAcct account date
+	 * @param costDetail optional cost detail - use to retrieve the cost history
+	 * @param trxName transaction name
+	 * @return MCost or null
+	 */
+	public static MCost get (Properties ctx, int AD_Client_ID, int AD_Org_ID, int M_Product_ID,
+			int M_CostType_ID, int C_AcctSchema_ID, int M_CostElement_ID,
+			int M_AttributeSetInstance_ID, 
+			Timestamp dateAcct, MCostDetail costDetail,
+			String trxName)

```

Comment:
The design of this new methods is very confusing. Is the intention here is always to get history cost when either dateAcct or costDetail is not null ? If that's the case, we should not mix it with the existing get(...) method that's always means to get the current MCost record. The new additions just make the usage of the get(...) method much more complicated and confusing.

Reply:
I will rename the method to getFromCostHistory(...)